extension/src/state_aggregate.rs

```diff
@@ -71,6 +110,85 @@ pub mod toolkit_experimental {
             let end = record.state_end as usize;
             &self.states_as_str()[beg..end]
         }
+
+        #[allow(clippy::unnecessary_unwrap)] // following clippy would make this much more convoluted 
```

Comment:
This is mostly correct, (None, Some(next)) is also valid.

Reply:
I fixed the clippy complaint as you mentioned, great suggestion.

For the `interpolate_first/last` I found that the code I was trying to refactor out wasn't well encapsulated due to the need to potentially add to the states string that was being built up for the result.  I left it as it is for now, since creating a `interpolate_start_and_update_state_string` doesn't seem much of an improvement, but it does seem like this could use a refactor.